dashboard/src/components/ErrorAlert/ErrorSelector.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from "react";
+import { definedNamespaces } from "../../shared/Namespace";
+import {
+  AppConflict,
+  ForbiddenError,
+  IRBACRole,
+  NotFoundError,
+  UnprocessableEntity,
+} from "../../shared/types";
+import { PermissionsErrorAlert, UnexpectedErrorAlert } from "./";
+import { namespaceText } from "./helpers";
+
+interface IErrorSelectorProps {
+  error: Error;
+  // Action related to the error (e.g. create, delete)
+  action?: string;
+  // Default permissions that the component should have
+  // returned in case it's a ForbiddenError and the message
+  // doesn't contain specific RBAC roles
+  defaultRequiredRBACRoles?: {
+    [action: string]: IRBACRole[];
+  };
+  // Namespace in case the error is namespaced
+  namespace?: string;
+  // Resource causing the failure (e.g. "Application my-wordpress")
+  resource?: string;
+}
+
+class ErrorSelector extends React.Component<IErrorSelectorProps> {
+  public render() {
+    const { namespace, error, defaultRequiredRBACRoles, action, resource } = this.props;
+    switch (error.constructor) {
+      case AppConflict:
```

Comment:
It seems that we are treating `AppConflict` like a generic error type when in reality it does not seem to be. 

I am wondering if we should remove this one from here and just allow `ErrorSelector` to support a message string.